### DISCUSSION TITLE
Is there any newbie friendly development guide?

### DISCUSSION THREAD

**Discussion by Davidliang11292** [Tuxemon/Tuxemon#2308] [General]
A while back, I was going to make a Tuxemon game, now I'm going to start over, and I think it's always a good idea to write a newbie friendly development guide. What do you think? Any comments and suggestions are welcome!

**Comment by HippasusTwo** [comment]
Not that I've seen but the event reference is a good starting place:
https://wiki.tuxemon.org/Event_Reference
oh and this
https://wiki.tuxemon.org/index.php?title=Map_Editor_Events
The key thing is most of the changes you'll want to make will be to files in 'mods', as that's where the maps are. The other key thing that took me by surprise is that debug.yaml is not just for debugging but also for the very start of a campaign and gender choice. If you wanted your campaign scenario to have only a certain character instead of customization, for instance, you'd set that there in the intro for your campaign (just be careful not to alter the others by accident).

**Reply by Davidliang11292** [reply]
> Not that I've seen but the event reference is a good starting place:
> https://wiki.tuxemon.org/Event_Reference
> oh and this
> https://wiki.tuxemon.org/index.php?title=Map_Editor_Events
> The key thing is most of the changes you'll want to make will be to files in 'mods', as that's where the maps are. The other key thing that took me by surprise is that debug.yaml is not just for debugging but also for the very start of a campaign and gender choice. If you wanted your campaign scenario to have only a certain character instead of customization, for instance, you'd set that there in the intro for your campaign (just be careful not to alter the others by accident).
> 

Thank you! This is a good document! But I thought it would be nice if Tuxemon had a sample project for teaching developers how to create games

**Reply by HippasusTwo** [reply]
Perhaps the Tuxemon Stadium campaign could be such a sample project. It doesn't exist yet though:
https://github.com/Tuxemon/Tuxemon/issues/2316

**Reply by Davidliang11292** [reply]
Thanks!
